scripts/superdesk-interfaces/Vocabulary.ts

```diff
@@ -16,10 +16,11 @@ export interface IVocabulary {
     priority: number;
     unique_field: string;
     schema: {};
-    field_type: 'text' | 'media' | 'date' | 'embed' | 'related_content';
+    field_type: 'text' | 'media' | 'date' | 'embed' | 'related_content' | 'custom';
     field_options: {};
     init_version: number;
     preffered_items: boolean;
     tags: Array<IVocabularyTag>;
     date_shortcuts: Array<{value: number; term: string; label: string}>;
+    custom_field_type: string;
```

Comment:
Are there any advantages to adding another field instead of converting `field_type` to string and storing custom field types there as well?

Reply:
that's for backward compatibility

Reply:
Could you explain more? What do you want this to be compatible with? Client repositories? Or you want to be able to implement this without having to refactor too much of server/client code. Would you need to refactor a lot if you wanted to avoid extra field and store additional field types in the existing field?

Reply:
actually it would probably work, not sure what I was thinking about. the only issue might be with the metadata settings UI where each new type has its own tab, and we show/hide some form fields based on it

Reply:
I don't see a reason why shouldn't we be able to generate the metadata tabs too. We can keep `custom_field_type` if it proves to be too much work to have it in the main field, but see if you can do that in a straight-forward way.

Reply:
Have you looked into this?

Reply:
I'm still not sure it would be better, instead of explicit check if `field_type === 'custom'` I would have to use `field_type not in ('text', 'media', 'urls', 'date', 'embed', 'related_content')` and there would be no type check on `field_type` possible

Reply:
but I would do some refactoring anyhow, eg those metadata tabs can be generated anyhow

Reply:
> I'm still not sure it would be better, instead of explicit check if field_type === 'custom' I would have to use field_type not in ('text', 'media', 'urls', 'date', 'embed', 'related_content') and there would be no type check on field_type possible

The idea I have for the future of this is to remove the difference between custom and non-custom fields. It doesn't have to be a separate extension for separate fields, but if it used the same interface, there would be no difference from the code point of view. It's easier to maintain when you need to work against a single interface rather than x types with different logic.

I've implemented such thing in [multi action bar](https://github.com/superdesk/superdesk-client-core/pull/2989/files#diff-80f5f9652545a6cdc07d3240a7a4b60aR32). There is an interface to define an action for multiple items that extensions can implement, but I also moved existing items on the same interface.

Reply:
yep that's something I'm also having in mind, all fields should use the same api